internal/controller/validatorconfig_controller.go

```diff
@@ -116,18 +131,49 @@ func (r *ValidatorConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-// updateStatus updates the ValidatorConfig's status subresource
-func (r *ValidatorConfigReconciler) updateStatus(ctx context.Context, vc *v1alpha1.ValidatorConfig) {
-	if err := r.Status().Update(context.Background(), vc); err != nil {
-		r.Log.V(0).Error(err, "failed to update ValidatorConfig status")
+// updateVc updates the ValidatorConfig object
+func (r *ValidatorConfigReconciler) updateVc(ctx context.Context) error {
+	if err := r.Get(ctx, vcKey, vc); err != nil {
+		r.Log.V(0).Error(err, "failed to get ValidatorConfig")
+		return err
+	}
+
+	// ensure cleanup finalizer
+	ensureFinalizer(ctx, r.Client, vc, CleanupFinalizer)
+	r.Log.V(0).Info("Ensured ValidatorConfig finalizer")
+
+	vc.Annotations = annotations
+
+	if err := r.Client.Update(ctx, vc); err != nil {
+		r.Log.V(1).Info("warning: failed to update ValidatorConfig annotations", "error", err)
```

Comment:
curious why this isn't a case where you'd want to return an error and retry

Reply:
resolved in 9d352385eed91878e6e463b08b54d6a79e5fc20d